ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -130,6 +134,16 @@ data Handlers m peer blk = Handlers {
         -> peer
         -> TxSubmissionServerPipelined (GenTxId blk) (GenTx blk) m ()
 
+    , hKeepAliveClient
+        :: NodeToNodeVersion
+        -> StdGen
+        -> ScheduledStop m
+        -> peer
+        -> (StrictTVar m (M.Map peer PeerGSV))
+        -> KeepAliveInterval
+        -> StrictTVar m (Maybe Time)
+        -> KeepAliveClient m ()
```

Comment:
Does it have to take all these arguments? Compared to the other handlers, it takes a lot of them. I would expect that the only extra arguments needed are the ones shared with other handlers. The other ones can be given in `mkHandlers`. For example, `ScheduledStop m` and `KeepAliveInterval` are just static and can be moved from `aKeepAliveClient` to `hKeepAliveClient` (or are you planning on making them dynamic or shared?)

Reply:
`ScheduledStop m` will be  used by all protocols when the new connection manager is merged. The keepalive protocol was developed on  top of that branch, which is why it already has it.
Do you have `aKeepAliveClient` and  `hKeepAliveClient` mixed up? 
Anyway the root of cause of `hKeepAliveClient` arguments' is `Ouroboros.Network.KeepAlive.keepAliveClient` and moving arguments between `aKeepAliveClient` and `hKeepAliveClient` wont change that.